SpatialGDK/Source/SpatialGDK/Private/Interop/WellKnownEntitySystem.cpp

```diff
@@ -50,14 +53,48 @@ void WellKnownEntitySystem::Advance()
 			break;
 		}
 	}
+
+	const FSubViewDelta& SubViewDelta = SystemEntitySubView->GetViewDelta();
+	for (const EntityDelta& Delta : SubViewDelta.EntityDeltas)
+	{
+		switch (Delta.Type)
+		{
+		case EntityDelta::REMOVE:
+			// If the snapshot partition authoritative server crashed, every other server checks to see if
+			// it should be the new leader, and thus claim the snapshot partition.
+			if (SnapshotPartitionServerSystemEntity == Delta.EntityId)
+			{
+				UE_LOG(LogWellKnownEntitySystem, Warning, TEXT("Server detected that snapshot partition auth server crashed."));
+				SnapshotPartitionAuthServerCrashed = true;
+				MaybeClaimSnapshotPartition();
+			}
+			// If we're the VTM auth server, and some other worker disconnected, we should check if it's a server
+			// worker, and if so, update the translation accordingly (so we're ready for the server to restart).
+			else if (VirtualWorkerTranslationManager != nullptr)
+			{
+				VirtualWorkerTranslationManager->OnWorkerDisconnected(Delta.EntityId);
+			}
+		default:
+			break;
+		}
+	}
 }
 
-void WellKnownEntitySystem::ProcessComponentUpdate(const Worker_ComponentId ComponentId, Schema_ComponentUpdate* Update)
+void WellKnownEntitySystem::ProcessComponentUpdate(const Worker_ComponentId EntityId, const Worker_ComponentId ComponentId,
+												   Schema_ComponentUpdate* Update)
 {
 	switch (ComponentId)
 	{
+	case SpatialConstants::PARTITION_COMPONENT_ID:
```

Comment:
Shouldn't we process this on EntityAdd also, in the event the partition was claimed whilst the server wasn't interested?

Reply:
Oh you are - nothing to see here